core/src/test/java/org/elasticsearch/index/query/PrefixQueryBuilderTest.java

```diff
@@ -0,0 +1,134 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.MultiTermQuery;
+import org.apache.lucene.search.PrefixQuery;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.query.support.QueryParsers;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.is;
+
+public class PrefixQueryBuilderTest extends BaseQueryTestCase<PrefixQueryBuilder> {
+
+    private static final String[] REWRITE_METHOD = new String[]{"constant_score_auto", "scoring_boolean", "constant_score_boolean",
+            "constant_score_filter", "top_terms_boost_1", "top_terms_1"};
+
+    @Override
+    protected PrefixQueryBuilder doCreateTestQueryBuilder() {
+        PrefixQueryBuilder query = newRandomQueryBuilder();
+
+        if (randomBoolean()) {
+            query.rewrite(randomFrom(REWRITE_METHOD));
+        }
+
+        return query;
+    }
+
+    private PrefixQueryBuilder newRandomQueryBuilder() {
+        String fieldName = null;
+        Object value;
+        switch (randomIntBetween(0, 3)) {
+            case 0:
+                if (randomBoolean()) {
+                    fieldName = STRING_FIELD_NAME;
+                }
+                if (frequently()) {
+                    value = randomAsciiOfLengthBetween(1, 10);
+                } else {
+                    // generate unicode string in 10% of cases
+                    value = randomUnicodeOfLength(10);
+                }
+                break;
+            case 1:
+                if (randomBoolean()) {
+                    fieldName = INT_FIELD_NAME;
+                }
+                value = randomInt(10000);
+                break;
+            case 2:
+                if (randomBoolean()) {
+                    fieldName = DOUBLE_FIELD_NAME;
+                }
+                value = randomDouble();
+                break;
+            case 3:
+                if (randomBoolean()) {
+                    fieldName = DATE_FIELD_NAME;
+                }
+                value = randomDate();
+                break;
+            default:
```

Comment:
This can't happen, and if for some reason after the switch fielname/value are unassigned the returned query will throw exceptions at some point (validation, doXContent, toQuery), so given this is a test method I'd remove it.